src/coordinator/services/m3coordinator/server/server.go

```diff
@@ -172,21 +172,31 @@ func setupStorages(logger *zap.Logger, session client.Session, cfg config.Config
 	cleanup := func() {}
 	localStorage := local.NewStorage(session, namespace)
 	stores := []storage.Storage{localStorage}
+	remoteEnabled := false
 	if cfg.RPC != nil && cfg.RPC.Enabled {
 		logger.Info("rpc enabled")
 		server := startGrpcServer(logger, localStorage, cfg.RPC)
 		cleanup = func() {
 			server.GracefulStop()
 		}
+
 		if remotes := cfg.RPC.RemoteListenAddresses; len(remotes) > 0 {
 			client, err := tsdbRemote.NewGrpcClient(remotes)
 			if err != nil {
 				logger.Fatal("unable to start remote clients for addresses", zap.Any("error", err))
 			}
+
 			stores = append(stores, remote.NewStorage(client))
+			remoteEnabled = true
 		}
 	}
-	fanoutStorage := fanout.NewStorage(stores, filter.LocalOnly, filter.LocalOnly)
+
+	readFilter := filter.LocalOnly
+	if remoteEnabled {
```

Comment:
You don't need the extra variable, can set`readFilter = filter.AllowAll` on line 190 where you're setting `remoteEnabled = true`

Reply:
separating the two logical paths. on line 190, remote being enabled, shouldn't necessarily tie to the filter of choice